### PR TITLE
11 principles page improvements

### PR DESCRIPTION
**Pull request by melindafekete**
Optimize the 11 principles page based on [Ercule feedback](https://docs.google.com/document/d/1aCf0leHvrB9sVjeG3N0icPcUo7y5FcgZz03qX5JvL1Y/edit?tab=t.0). Key changes:
- Reorganize the principles based on heatmap data
- Add more links
- Add more screenshots, videos and a table for a visual overview
- Add additional resources section

[Page preview](https://unleash-docs-git-melinda-11-principles-updates-unleash-team.vercel.app/topics/feature-flags/feature-flag-best-practices)

### CODE REVIEW

File: website/docs/topics/feature-flags/feature-flag-best-practices.mdx [hunk 1]
```diff
@@ -85,11 +171,13 @@ For optimal performance, you should evaluate feature flags as close to your user
 
 In summary, this principle emphasizes optimizing performance while protecting end-user privacy by evaluating feature flags as close to the end user as possible. This also leads to a highly available feature flag system that scales with your applications.
 
-## 4. Scale horizontally by decoupling reads and writes
+<VideoContent videoUrls={["https://www.youtube.com/embed/mCXSAWzdn3I"]}/>
```
Comment (alvinometric) [inline website/docs/topics/feature-flags/feature-flag-best-practices.mdx:174]: Not a big fan of that video here. I know we don't have a lot of choice, so maybe let's use this as an opportunity to make one? Or ask the team who wrote it to do that

File: website/docs/topics/feature-flags/feature-flag-best-practices.mdx [hunk 2]
```diff
@@ -191,4 +229,20 @@ To ensure a good developer experience, you should provide the following:
...
 - **Documentation and training**: Offer clear, comprehensive documentation for the API, UI, and SDKs, with easy-to-follow examples. This simplifies onboarding for new developers and supports continuous training, ensuring the effective use of the feature flagging system.
 
-Thank you for reading. Our motivation for writing these principles is to share what we've learned building a large-scale feature flag solution with other architects and engineers solving similar challenges. Unleash is open-source, and so are these principles. Have something to contribute? [Open a PR](https://github.com/Unleash/unleash/pulls) or [discussion](https://github.com/orgs/Unleash/discussions) on our GitHub.
+---
+
+## Additional resources
+
+Thank you for reading. Our motivation for writing these principles is to share what we've learned building a large-scale feature flag solution with other architects and engineers solving similar challenges. Unleash is open-source, and so are these principles.
+Have something to contribute? [Open a PR](https://github.com/Unleash/unleash/pulls) or [discussion](https://github.com/orgs/Unleash/discussions) on our GitHub.
+
```
Comment (alvinometric) [inline website/docs/topics/feature-flags/feature-flag-best-practices.mdx:238]: I would get rid of everything after this line

I'm not sure we want to give people even more information that they haven't asked for, but we could put a link to our pricing page if they're curious

File: website/docs/topics/feature-flags/feature-flag-best-practices.mdx [hunk 3]
```diff
@@ -41,25 +41,113 @@ Let's break down these components.
...
+## 3. Make flags short-lived
+
+The most common use case for feature flags is to manage the [rollout](https://www.getunleash.io/feature-flag-use-cases-progressive-or-gradual-rollouts) of new functionality. Once a rollout is complete, you should remove the feature flag from your code and archive it. Remove any old code paths that the new functionality replaces.
+
+Avoid using feature flags for static application configuration. Application configuration should be consistent, long-lived, and loaded during application startup. In contrast, feature flags should be short-lived, dynamic, and updated at runtime. They prioritize availability over consistency and can be modified frequently.
+
+<table>
```
Comment (alvinometric) [inline website/docs/topics/feature-flags/feature-flag-best-practices.mdx:65]: Did you prefer an HTML table just for ease of reading?

**Review by alvinometric — APPROVED** [review]
Good work updating all the anchors 😄

File: website/docs/topics/feature-flags/feature-flag-best-practices.mdx (hunk 1 above)
Comment (melindafekete) [reply website/docs/topics/feature-flags/feature-flag-best-practices.mdx:174]: Sorry, this is in the wrong place, it's meant to go with the SDKs section, just got tangled up during a merge conflict

File: website/docs/topics/feature-flags/feature-flag-best-practices.mdx (hunk 2 above)
Comment (melindafekete) [reply website/docs/topics/feature-flags/feature-flag-best-practices.mdx:238]: It was a suggestion from Ercule to add this section on additional resources: https://docs.google.com/document/d/1aCf0leHvrB9sVjeG3N0icPcUo7y5FcgZz03qX5JvL1Y/edit?tab=t.0

Given it's a page that drives a lot of traffic, I think it makes sense to give people some ideas of where to go next to learn more. The pricing page feels a bit too direct

File: website/docs/topics/feature-flags/feature-flag-best-practices.mdx [hunk 4]
```diff
@@ -71,11 +159,11 @@ In [Principle 1](#1-enable-runtime-control), we proposed a set of architectural
 
 For client-side feature flags, SDKs should send the context to an evaluation server and receive the evaluated results. The evaluated results are then cached in memory in the client-side application, allowing quick lookups without additional network overhead. This provides the performance benefits of local evaluation while minimizing the exposure of sensitive data.
 
-This approach enhances privacy by minimizing the amount of sensitive data sent to the _Feature Flag Control Service_, reducing the risk of data exposure and potential security threats.
+<VideoContent videoUrls={["https://www.youtube.com/embed/mCXSAWzdn3I"]}/>
```
Comment (melindafekete) [inline website/docs/topics/feature-flags/feature-flag-best-practices.mdx:162]: @alvinometric do you think the video makes more sense here, or do you still have doubts?

**Comment by ferrantim** [comment]
LGTM, thank you @melindafekete

**Pull request merged**